app/controllers/merchant/discounts_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Merchant::DiscountsController < Merchant::BaseController
+
+  def index
+    @discounts = Discount.all
+  end
+
+  def new
+    merchant = Merchant.find_by(id: current_user[:merchant_id])
+    @items = merchant.items
+  end
+
+  def create
+    params[:merchant_id] = current_user.merchant_id
+    discount = Discount.create(discount_params)
+    if discount.save
+      item_ids = params[:applicable_items].values.flatten.reject! {|id| id.empty? }
+      if item_ids.empty?
+        Merchant.find(current_user[:merchant_id]).items.each do |item|
+          discount.discount_items.create ({ discount_id: discount.id, item_id: item.id })
+        end
+      else
+        item_ids.each do |item_id|
+          discount.discount_items.create ({ discount_id: discount.id, item_id: item_id })
+        end
+      end
+      redirect_to "/merchant/discounts"
+    else
+      flash[:notice] = "Percent discount is required as well as either minimum quantity or minimum value!"
+      redirect_to "/merchant/discounts/new"
+    end
+  end
```

Comment:
This method absolutely needs to be refactored. At the moment I'm having trouble imagining how to get it out of this controller and either into a discount_items controller or the model because of the need to access the params in order to be able to determine actions to take. Therefore, putting this in the "functional but needs to be refactored" category.